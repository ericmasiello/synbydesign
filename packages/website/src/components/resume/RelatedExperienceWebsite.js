import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './RelatedExperienceWebsite.module.css';

function RelatedExperienceWebsite(props) {
  const { as: Component, className, ...rest } = props;
  const classes = classNames(styles.experience, className);
  return <Component className={classes} {...rest} />;
}

RelatedExperienceWebsite.defaultProps = {
  as: 'div',
};

RelatedExperienceWebsite.propTypes = {
  as: PropTypes.elementType,
};

export default RelatedExperienceWebsite;
