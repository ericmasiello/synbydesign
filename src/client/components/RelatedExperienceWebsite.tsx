import * as React from 'react';
import styled from 'styled-components';

interface Props extends Website {
  className?: string;
}

interface DefaultProps {
  disabled: boolean;
}

export const RelatedExperienceWebsite: React.SFC<Props> = props => {
  const { title, url, disabled, ...rest } = props as Props & DefaultProps;

  if (disabled) {
    return <span {...rest}>{title || url}</span>;
  }
  return (
    <a href={url} {...rest}>
      {title || url}
    </a>
  );
};

RelatedExperienceWebsite.defaultProps = {
  disabled: false,
} as DefaultProps;

RelatedExperienceWebsite.displayName = 'RelatedExperience.Website';

export default styled(RelatedExperienceWebsite)``;
