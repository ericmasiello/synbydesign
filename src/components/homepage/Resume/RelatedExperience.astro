---
import classNames from 'classnames';
import Text from '../../Text.astro';
import PlainList from '../../PlainList.astro';
import type { RelatedExperience } from './Resume';
import Spacer from './Spacer.astro';
import type { Props as SpacerProps } from './Spacer.astro';

export interface Props extends Omit<SpacerProps, 'slot' | 'title' | 'role'>, RelatedExperience {}

const { as = 'article', class: className, title, meta, role, website, accomplishments, ...rest } = Astro.props;
const classes = classNames('experience', className);
const years = role.yearFrom && role.yearTo ? `${role.yearFrom} â€” ${role.yearTo}` : role.yearFrom;
---

<Spacer class={classes} {...rest} as={as}>
  <hgroup>
    <Text as="h4" data-title class="title">
      <span class="title-content" data-title-content set:html={title}></span>{' '}
      {
        years && (
          <Text small muted>
            ({years})
          </Text>
        )
      }
    </Text>
    {
      meta && (
        <Text small as="p" data-meta class="meta">
          {meta}
        </Text>
      )
    }
    <Text as="h5" data-role-title class="role">
      {role.title}
    </Text>
    {
      website && (
        <a href={website.url} class="website">
          {website.url}
        </a>
      )
    }
  </hgroup>
  {
    accomplishments.length > 0 && (
      <div>
        <PlainList>
          {accomplishments.map((accomplishment) => (
            <li>{accomplishment}</li>
          ))}
        </PlainList>
      </div>
    )
  }
</Spacer>
<style>
  .experience:not(:last-of-type) {
    page-break-before: left;
  }

  .title {
    margin-block-end: 0.125rem;
  }

  .title-content {
    font-weight: var(--font-header-weight-bold);
  }

  .meta {
    margin-block-end: 0;
  }

  .role {
    color: var(--color-brand);
    margin-block-end: 0.5rem;
  }

  .website {
    display: block;
    margin-block-end: 0.5rem;
  }
</style>
