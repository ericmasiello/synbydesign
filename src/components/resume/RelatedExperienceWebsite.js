import React from 'react';
import PropTypes from 'prop-types';

function RelatedExperienceWebsite(props) {
  const { as: Component, className, ...rest } = props;
  return <Component className={className} {...rest} />;
}

RelatedExperienceWebsite.defaultProps = {
  as: 'div',
};

RelatedExperienceWebsite.propTypes = {
  as: PropTypes.elementType,
};

export default RelatedExperienceWebsite;
