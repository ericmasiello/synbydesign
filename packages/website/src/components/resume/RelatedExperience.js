import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import PlainList from '../PlainList';
import { Text, Muted } from '../Type';
import RelatedExperienceWebsite from './RelatedExperienceWebsite';
import styles from './RelatedExperience.module.css';

function RelatedExperience(props) {
  const { as: Component, className, title, meta, role, website, accomplishments, ...rest } = props;
  const classes = classNames(styles.experience, className);
  const years = role.yearFrom && role.yearTo ? `${role.yearFrom} â€” ${role.yearTo}` : role.yearFrom;
  return (
    <Component className={classes} {...rest}>
      <hgroup>
        <Text as="h4" data-title className={styles.title}>
          <span className={styles.titleContent} data-title-content dangerouslySetInnerHTML={{ __html: title }} />{' '}
          {years && (
            <Text small as={Muted}>
              ({years})
            </Text>
          )}
        </Text>
        {meta && (
          <Text small as="p" data-meta className={styles.meta}>
            {meta}
          </Text>
        )}
        <Text as="h5" data-role-title className={styles.role}>
          {role.title}
        </Text>
        {website && <RelatedExperienceWebsite {...website} className={styles.website} />}
      </hgroup>
      {accomplishments.length > 0 && (
        <section>
          <PlainList>
            {accomplishments.map((accomplishment) => (
              <li key={accomplishment}>{accomplishment}</li>
            ))}
          </PlainList>
        </section>
      )}
    </Component>
  );
}

RelatedExperience.defaultProps = {
  as: 'article',
};

RelatedExperience.propTypes = {
  as: PropTypes.elementType,
};

export default RelatedExperience;
