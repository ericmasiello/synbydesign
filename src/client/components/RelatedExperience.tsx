import * as React from 'react';
import styled from 'styled-components';
import RelatedExperienceWebsite from './RelatedExperienceWebsite';
import PlainList from './PlainList';
import TypeBase from './TypeBase';
import TypeSmall from './TypeSmall';
import Muted from './Muted';
import { COLORS, HEADER_WEIGHTS } from '../styles/vars';

export interface Props extends RelatedExperience {
  className?: string;
  tag?: Tag;
}

interface DefaultProps {
  tag: Tag;
  accomplishments: string[];
}

export const RelatedExperience: React.SFC<Props> = props => {
  const {
    tag: Tag,
    title,
    meta,
    role,
    website,
    accomplishments,
    className,
    ...rest
  } = props as Props & DefaultProps;
  return (
    <Tag className={className} {...rest}>
      <hgroup>
        <TypeBase tag="h1" data-title className="related-experience__title">
          <span
            className="related-experience__title__content"
            data-title-content
            dangerouslySetInnerHTML={{ __html: title }}
          />{' '}
          {role.yearFrom && (
            <TypeSmall tag={Muted}>({role.yearFrom})</TypeSmall>
          )}
        </TypeBase>
        {meta && (
          <TypeSmall tag="p" data-meta className="related-experience__meta">
            {meta}
          </TypeSmall>
        )}
        <TypeBase tag="h2" data-role-title className="related-experience__role">
          {role.title}
        </TypeBase>
        {website && (
          <RelatedExperienceWebsite
            {...website}
            className="related-experience__website"
          />
        )}
      </hgroup>
      {accomplishments.length > 0 && (
        <section>
          <PlainList>
            {accomplishments.map(accomplishment => (
              <li key={accomplishment}>{accomplishment}</li>
            ))}
          </PlainList>
        </section>
      )}
    </Tag>
  );
};

RelatedExperience.defaultProps = {
  tag: 'article',
  accomplishments: [],
} as DefaultProps;

RelatedExperience.displayName = 'RelatedExperience';

export default styled(RelatedExperience)`
  &:not(:last-of-type) {
    margin-bottom: 2rem;
  }

  .related-experience__title {
    margin-bottom: 0;
  }

  .related-experience__title__content {
    font-weight: ${HEADER_WEIGHTS.bold};
  }

  .related-experience__meta {
    margin-bottom: 0;
  }

  .related-experience__role {
    color: ${COLORS.highlight};
    margin-bottom: 0.5rem;
  }

  .related-experience__website {
    display: block;
    margin-bottom: 0.5rem;
  }
`;
