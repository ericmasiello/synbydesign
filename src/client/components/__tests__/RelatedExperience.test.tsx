import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { RelatedExperience, Props } from '../RelatedExperience';
import RelatedExperienceWebsite from '../RelatedExperienceWebsite';

const base: Props = {
  title: 'Title',
  role: {
    title: 'Role title',
  },
  accomplishments: [],
};

it('should render', () => {
  const component = renderer.create(<RelatedExperience {...base} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('should display additional classes', () => {
  const wrapper = shallow(<RelatedExperience {...base} className="my-class" />);

  expect(wrapper.hasClass('my-class')).toBe(true);
});

it('should render title', () => {
  const wrapper = shallow(<RelatedExperience {...base} />);

  expect(
    wrapper.find('h1 > span').props().dangerouslySetInnerHTML!.__html,
  ).toBe(base.title);
});

it('should render title as html', () => {
  const wrapper = shallow(
    <RelatedExperience {...base} title={'The <a href="#">link</a>'} />,
  );

  expect(
    wrapper.find('h1 > span').props().dangerouslySetInnerHTML!.__html,
  ).toBe('The <a href="#">link</a>');
});

it('should render meta data', () => {
  const props: Props = {
    ...base,
    meta: 'The meta',
  };
  const wrapper = shallow(<RelatedExperience {...props} />);

  expect(wrapper.find('p').text()).toContain('The meta');
});

it('should render the role starting year', () => {
  const props: Props = {
    ...base,
    role: {
      ...base.role,
      yearFrom: '2018',
    },
  };
  const wrapper = shallow(<RelatedExperience {...props} />);

  expect(wrapper.find('h1').text()).toContain('(2018)');
});

it('should render the role title', () => {
  const wrapper = shallow(<RelatedExperience {...base} />);

  expect(wrapper.find('h2').text()).toContain(base.role.title);
});

it('should render the website', () => {
  const props: Props = {
    ...base,
    website: {
      url: 'http://www.synbydesign.com',
      title: 'Syn By Design',
      disabled: true,
    },
  };
  const wrapper = shallow(<RelatedExperience {...props} />);
  const website = wrapper.find(RelatedExperienceWebsite);

  expect(website.props()).toEqual(props.website);
});

it('should render accomplishments', () => {
  const accomplishments = ['foo', 'bar', 'baz'];
  const wrapper = shallow(
    <RelatedExperience {...base} accomplishments={accomplishments} />,
  );

  accomplishments.forEach(accomplishment => {
    expect(wrapper.containsMatchingElement(<li>{accomplishment}</li>)).toBe(
      true,
    );
  });
});
