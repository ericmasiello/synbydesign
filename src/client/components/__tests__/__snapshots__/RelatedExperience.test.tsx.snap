// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<article
  className={undefined}
>
  <hgroup>
    <h1>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "Title",
          }
        }
      />
       
    </h1>
    <h2>
      Role title
    </h2>
  </hgroup>
</article>
`;
